modules/disasterRecovery.js

```diff
@@ -119,4 +120,79 @@ export function checkForControlPlaneSla(clusterDetails) {
     severity: Severity.High,
     details: details
   }
+}
+
+//
+// Checks if replication is enabled on container registries
+//
+export async function checkForContainerRegistryReplication(containerRegistries) {
+
+  console.log(chalk.white("Checking for container registry replication..."));
+
+  let details = []
+
+  // If there are no container registries the test does not apply
+  if (!containerRegistries.length) {
+    details.push({
+      status: ResultStatus.NotApply,
+      message: "No registries were specified. Skipping check"
+    });
+
+    return {
+      checkId: 'DR-7',
+      status: ResultStatus.NotApply,
+      severity: Severity.Medium,
+      details: details
+    }
+  }
+
+  // Figure out the registries that do not have replication
+  var registriesWithoutReplication = [];
+  for (const registry of containerRegistries) {
+
+    // Non-premium registries are always without replication
+    if (!equalsIgnoreCase(registry.sku.name, 'premium')) {
+      registriesWithoutReplication.push(registry.name);
+      continue;
+    }
+
+    // For premium registries we need to query replications
+    try {
+      var commandResults = await executeCommand(`az acr replication list -r ${registry.name}`);
+      var registryReplications = JSON.parse(commandResults.stdout);
+      if (registryReplications.length <= 1)
+        registriesWithoutReplication.push(registry.name);
+    }
+    catch (e) {
+      console.log(chalk.red(`--- An error occurred retrieving ACR replications: ${e}`));
```

Comment:
Could we just return with an error and not use `console.log`?

Reply:
Done.